src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -0,0 +1,324 @@
+    private final TalonFX shoulderAuxMotor;
+    private final TalonFX elbowMainMotor;
+    private final TalonFX elbowAuxMotor;
+    private final DutyCycleEncoder shoulderEncoder;
+    private final DutyCycleEncoder elbowEncoder;
+
+
+    private static ArmIOReal INSTANCE = null;
+
+    private final ArmKinematics kinematics = new ArmKinematics(ArmConstants.SHOULDER_ARM_LENGTH, ArmConstants.ELBOW_ARM_LENGTH);
+
+    private double shoulderOffset = ArmConstants.SHOULDER_ABSOLUTE_ENCODER_OFFSET;
+    private double elbowOffset = ArmConstants.ELBOW_ABSOLUTE_ENCODER_OFFSET;
+
+    private double shoulderFeedforward;
+    private double elbowFeedforward;
+
+    private Command lastCommand;
+    private boolean changedToDefaultCommand = false;
+
+    private double ySetpoint = 0;
+
+    private ControlRequest motorControlRequest = null;
+
+
+
+
+    ArmIOReal() {
+
+        this.shoulderEncoder = new DutyCycleEncoder(Ports.ArmPorts.SHOULDER_ENCODER);
+        this.elbowEncoder = new DutyCycleEncoder(Ports.ArmPorts.ELBOW_ENCODER);
+
+        this.shoulderMainMotor = new TalonFX(Ports.ArmPorts.SHOULDER_MAIN_MOTOR);
+
+        this.shoulderAuxMotor = new TalonFX(Ports.ArmPorts.SHOULDER_AUX_MOTOR);
+
+        this.elbowMainMotor = new TalonFX(Ports.ArmPorts.ELBOW_MAIN_MOTOR);
+
+        this.elbowAuxMotor = new TalonFX(Ports.ArmPorts.ELBOW_AUX_MOTOR);
+
+
+
+        shoulderAuxMotor.setControl(new StrictFollower(shoulderMainMotor.getDeviceID()));
+        elbowAuxMotor.setControl(new StrictFollower(elbowMainMotor.getDeviceID()));
+
+
+        shoulderMainMotorConfig.Voltage.PeakForwardVoltage = ArmConstants.VOLT_COMP_SATURATION;
+        shoulderMainMotorConfig.Voltage.PeakReverseVoltage = -ArmConstants.VOLT_COMP_SATURATION;
+        shoulderMainMotorConfig.Slot0 = SHOULDER_PID;
+        elbowMainMotorConfig.CurrentLimits.SupplyCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.CurrentLimits.StatorCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        shoulderMainMotorConfig.Feedback.SensorToMechanismRatio = SENSOR_TO_MECHANISM_RATIO;
+        shoulderMainMotorConfig.Feedback.RotorToSensorRatio = ROTOR_TO_SENSOR_RATIO;
+        shoulderMainMotor.getConfigurator().apply(shoulderMainMotorConfig);
+
+        shoulderMainMotor.setNeutralMode(NeutralModeValue.Brake);
+        shoulderMainMotor.setInverted(MAIN_CLOCKWISE);
+
+        elbowMainMotorConfig.Voltage.PeakForwardVoltage = ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Voltage.PeakReverseVoltage = -ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Slot1 = ELBOW_PID;
+        elbowMainMotorConfig.CurrentLimits.SupplyCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.CurrentLimits.StatorCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.Feedback.SensorToMechanismRatio = SENSOR_TO_MECHANISM_RATIO;
+        elbowMainMotorConfig.Feedback.RotorToSensorRatio = ROTOR_TO_SENSOR_RATIO;
```

Comment:
All of that should be in the constants file. Use the `with*` methods and remove the constants, this will be cleaner and shorter -- you won't need one line to define the constant and another to use it, you'll just have one line total.